vm/src/vm/vm_ops.rs

```diff
@@ -103,371 +125,217 @@ impl VirtualMachine {
-        default: &'static PyStrInterned,
-        reflection: &'static PyStrInterned,
-        unsupported: fn(&VirtualMachine, &PyObject, &PyObject) -> PyResult,
+        a: &PyObject,
+        b: &PyObject,
+        iop_slot: &PyNumberBinaryOpSlot,
+        op_slot: &PyNumberBinaryOpSlot,
     ) -> PyResult {
-        if rhs.fast_isinstance(lhs.class()) {
-            let lop = lhs.get_class_attr(reflection);
-            let rop = rhs.get_class_attr(reflection);
-            if let Some((lop, rop)) = lop.zip(rop) {
-                if !lop.is(&rop) {
-                    if let Ok(r) = self.call_or_unsupported(rhs, lhs, reflection, |vm, _, _| {
-                        Err(vm.new_exception_empty(vm.ctx.exceptions.exception_type.to_owned()))
-                    }) {
-                        return Ok(r);
-                    }
-                }
+        if let Some(slot) = a.class().slots.number.get_left_binary_op(iop_slot)? {
+            let x = slot(a.to_number(), b, self)?;
+            if !x.is(&self.ctx.not_implemented) {
+                return Ok(x);
             }
         }
-        // Try to call the default method
-        self.call_or_unsupported(lhs, rhs, default, move |vm, lhs, rhs| {
-            // Try to call the reflection method
-            // don't call reflection method if operands are of the same type
-            if !lhs.class().is(rhs.class()) {
-                vm.call_or_unsupported(rhs, lhs, reflection, |_, rhs, lhs| {
-                    // switch them around again
-                    unsupported(vm, lhs, rhs)
-                })
-            } else {
-                unsupported(vm, lhs, rhs)
-            }
-        })
+        self.binary_op1(a, b, op_slot)
     }
 
-    pub fn _sub(&self, a: &PyObject, b: &PyObject) -> PyResult {
-        self.call_or_reflection(
-            a,
-            b,
-            identifier!(self, __sub__),
-            identifier!(self, __rsub__),
-            |vm, a, b| Err(vm.new_unsupported_binop_error(a, b, "-")),
-        )
-    }
-
-    pub fn _isub(&self, a: &PyObject, b: &PyObject) -> PyResult {
-        self.call_or_unsupported(a, b, identifier!(self, __isub__), |vm, a, b| {
-            vm.call_or_reflection(
-                a,
-                b,
-                identifier!(self, __sub__),
-                identifier!(self, __rsub__),
-                |vm, a, b| Err(vm.new_unsupported_binop_error(a, b, "-=")),
-            )
-        })
-    }
+    fn binary_iop(
+        &self,
+        a: &PyObject,
+        b: &PyObject,
+        iop_slot: &PyNumberBinaryOpSlot,
+        op_slot: &PyNumberBinaryOpSlot,
+        op: &str,
+    ) -> PyResult {
+        let result = self.binary_iop1(a, b, iop_slot, op_slot)?;
+        if !result.is(&self.ctx.not_implemented) {
+            return Ok(result);
+        }
+        Err(self.new_unsupported_binop_error(a, b, op))
+    }
+
+    binary_func!(_sub, Subtract, "-");
+    binary_func!(_mod, Remainder, "%");
```

Comment:
So.. maybe cpython have special way to handle this? or it came from the different implementation of binary op wrapper?